modules/gui/research.lua

```diff
@@ -33,30 +33,37 @@ local empty_time = format_time(0, {
 	null=true
 })
 
+local font_color = {
+    [1] = {r = 0.3, g = 1, b = 0.3},
+    [2] = {r = 1, g = 0.3, b = 0.3}
+}
+
 local res = {
 	['lookup_name'] = {},
 	['disp'] = {}
 }
 
-local res_total = 0
-local mi = 1
+local function research_init()
```

Comment:
Looks like you have made this into a function and changed when it is calculated. This isn't needed and to prevent scope issues you can so `do end` blocks.
```suggestion
do
```